src/jit/codegenlinear.cpp

```diff
@@ -113,6 +113,10 @@ void CodeGen::genInitialize()
     // Make sure a set is allocated for compiler->compCurLife (in the long case), so we can set it to empty without
     // allocation at the start of each basic block.
     VarSetOps::AssignNoCopy(compiler, compiler->compCurLife, VarSetOps::MakeEmpty(compiler));
+
+    // We initialize the stack level before fist "BasicBlock" code is generated in case we need to report stack
+    // variable needs home and so its stack offset.
+    SetStackLevel(0);
```

Comment:
Before we were initializing StackLevel in `CodeGen::genCodeForBBlist()`:
https://github.com/dotnet/coreclr/blob/8497ae3ed438f443b22b00a66f42afe8f6623bbf/src/jit/codegenlinear.cpp#L344-L348

Looks like now this call is not necessary and can be replaced with assert?

Reply:
We actually need this inside the for, because is not true that the stack level is set to 0 at the end of each block, but it should be use as that when generating code for a block.

Reply:
> because is not true that the stack level is set to 0 at the end of each block,
Hm, maybe we should fix it, I will take a look.

Reply:
I think @BruceForstall knows more about this case

Reply:
So we check that it is empty after each basic block in the same loop:
https://github.com/dotnet/coreclr/blob/69dc790d171783584f5af3a5da60a25e2c301899/src/jit/codegenlinear.cpp#L573-L574

and I was not able to hit that assert. So could you give me a repro where we still need it with your change? And do you still need this change after discussion in others PR about variable's home?

Reply:
I have opened [an issue](https://github.com/dotnet/coreclr/issues/23375) for that.